src/containers/NdlaFilm/NdlaFilmEditor.jsx

```diff
@@ -0,0 +1,315 @@
+      ...filmFrontpage,
+      slideShow: newSlideShow,
+    };
+    this.saveFilmFrontpage(newFilmFrontpage);
+  };
+
+  saveFilmFrontpage = newFilmFrontpage => {
+    updateFilmFrontpage(newFilmFrontpage);
+    this.refreshProps(newFilmFrontpage);
+  };
+
+  onSaveThemeName = (newNames, index) => {
+    const { themes } = this.state;
+    const oldTheme = themes[index];
+    const newThemeNames = this.convertThemeNames(newNames);
+
+    const newTheme = {
+      ...oldTheme,
+      name: newThemeNames,
+    };
+    this.updateThemeName(newTheme, index);
+  };
+
+  updateThemeName = (updatedTheme, index) => {
+    const movieIds = updatedTheme.movies.map(movie => getUrnFromId(movie.id));
+    const { filmFrontpage } = this.state;
+    const { movieThemes } = filmFrontpage;
+
+    const newFilmFrontpage = {
+      ...filmFrontpage,
+      movieThemes: movieThemes.map((theme, i) => {
+        if (i === index) {
+          return {
+            ...updatedTheme,
+            movies: movieIds,
+          };
+        }
+        return theme;
+      }),
+    };
+    this.saveFilmFrontpage(newFilmFrontpage);
+  };
+
+  updateMovieTheme = (updatedTheme, index) => {
+    const movieIds = updatedTheme.map(movie => getUrnFromId(movie.id));
+    const { filmFrontpage } = this.state;
+    const { movieThemes } = filmFrontpage;
+
+    const newFilmFrontpage = {
+      ...filmFrontpage,
+      movieThemes: movieThemes.map((theme, i) => {
+        if (i === index) {
+          return {
+            ...theme,
+            movies: movieIds,
+          };
+        }
+        return theme;
+      }),
+    };
+    this.saveFilmFrontpage(newFilmFrontpage);
+  };
+
+  addMovieToTheme = (id, index) => {
+    const { filmFrontpage } = this.state;
+    const urnId = getUrnFromId(id);
+
+    const newFilmFrontpage = {
+      ...filmFrontpage,
+      movieThemes: filmFrontpage.movieThemes.map((theme, i) => {
+        if (i === index) {
+          return {
+            ...theme,
+            movies: [...theme.movies, urnId],
+          };
+        }
+        return theme;
+      }),
+    };
+    this.saveFilmFrontpage(newFilmFrontpage);
+  };
+
+  onMoveTheme = (index, direction) => {
+    const { filmFrontpage } = this.state;
+    if (
+      index + direction >= 0 &&
+      filmFrontpage.movieThemes.length > index + direction
+    ) {
+      const desiredNewIndex = index + direction;
+      const { movieThemes } = filmFrontpage;
+
+      const newMovieThemes = movieThemes.map((movieTheme, i) => {
+        if (i === index) {
+          return movieThemes[desiredNewIndex];
+        } else if (i === desiredNewIndex) {
+          return movieThemes[index];
+        }
+        return movieTheme;
+      });
+
+      const newFilmFrontpage = {
+        ...filmFrontpage,
+        movieThemes: newMovieThemes,
+      };
+      this.saveFilmFrontpage(newFilmFrontpage);
+    }
+  };
+
+  onDeleteTheme = index => {
+    const { filmFrontpage } = this.state;
```

Comment:
Dette gjelder over alt hvor saveFilmFrontpage blant annet er brukt og this.state er brukt. Her burde det egentlig blitt brukt prevState.